packages/grafana-ui/src/components/Input/AutoSizeInput.tsx

```diff
@@ -20,14 +20,14 @@ export const AutoSizeInput = React.forwardRef<HTMLInputElement, Props>((props, r
   const [inputWidth, setInputWidth] = React.useState(minWidth);
 
   useEffect(() => {
-    setInputWidth(getWidthFor(value.toString(), minWidth, maxWidth));
+    setInputWidth(getWidthFor(value?.toString(), minWidth, maxWidth));
   }, [value, maxWidth, minWidth]);
 
   return (
     <Input
       {...restProps}
       ref={ref}
-      value={value.toString()}
+      value={value?.toString()}
```

Comment:
I want to hold off on this change for a little bit. 

The type of `value` here says it's _impossible_ for this to be a falsey value, therefor it's not intutive for future developers why `?.` has been used. Somewhere, some times need to be updated to actually fix this problem.

Reply:
Hey @joshhunt! So in the escalation I noticed that affected dashboards are somehow automatically generated and have `null` in values:
```json
"current": {
   "text": null, 
   "value": null
}

```
old arch dashboard also automatically updates that to have text and value to be `""` after load. In scenes it breaks because it does not convert that to `""`.  It also fixes `null` when importing dashboard, so I assume this would only affect programmatically created dashboards. Very niche situation

Reply:
The problem here is that someone will inevitably refactor this component, see that the optional-chaining operator is unnessicary, remove it, and reintroduce this bug.

`null` is an invalid value to pass into `AutoSizeInput` / `Input`. If you attempt to do that, you will get a build error telling you that you have a bug in your code

![7966_2024-08-01-13-13_Code](https://github.com/user-attachments/assets/3ceb0c47-e826-4d40-9217-ba942e580cc6)

From this vantage point, I would say that the consuming code is wrong and should be ensuring that null isn't passed in. It's unclear to me whether there's other issues that this invalid value might be causing?

I want to make sure we get the right pragmatic approach that:
1. fixes the problem more whole-y
2. doesn't cause future regressions

Do you have a link to the offending AutoSizeInput-consuming code?

Reply:
Yeah, should probably address it from scenes so that refactor doest break it. This is consumed in [scenes](https://github.com/grafana/scenes/blob/0889dd2335aceb5c3afd9b78127e5ce0e1a75008/packages/scenes/src/variables/components/VariableValueInput.tsx#L28), but data is handled in [main](https://github.com/grafana/grafana/blob/2e2ddc5c42a3ccb2a1982d00b3e102e5ff21f2e6/public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts#L389) so adjusting data handling makes more sense. Will close this PR and open another.